client/src/main/java/io/pravega/client/stream/Transaction.java

```diff
@@ -70,9 +70,11 @@
      * Send a transaction heartbeat and increase transaction's timeout by lease amount of milliseconds.
      *
      * @param lease Additional amount of time in milliseconds by which to increase transaction's timeout.
+     * @param switchOver Boolean indicating whether the previous successful ping request was performed on a different
+     *                   controller process by the client.
      * @throws PingFailedException Ping failed.
      */
-    void ping(long lease) throws PingFailedException;
+    void ping(long lease, boolean switchOver) throws PingFailedException;
```

Comment:
@fpj, yes, there is no notion of `switchover` in any ZK's ping API, and we could exploit zookeeper sessions for this purpose as you suggest. We could also employ @shiveshr's solution of having a ping request stream. I have filed a separate issue #1499 to discuss this problem further and resolve it.